app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -1057,6 +1068,13 @@ public class MainActivity : AppCompatActivity(), MainViewController, RouteCallba
         poiTapPoint = null
     }
 
+    override fun onSharedPreferenceChanged(preferences: SharedPreferences?, key: String?) {
```

Comment:
I guess we don't actually need a preference listener, I had implemented it that way so that we only make changes when the value of 'cache search results changes'. We can definitely do something like what you're suggesting and rely on updating it when `onResume()` is called. Since we need to called `invalidateOptionsMenu()` for `onPrepareOptionsMenu()` to be called, I've made `onResume()` call `invalidateOptionsMenu()` and put the search bar configuration code in `onPrepareOptionsMenu()` since it is part of the menu. How does that sound?